### PR TITLE
fix(sdk): Add theming options for Collection Browser's empty content message

### PR DESCRIPTION
**Pull request by oisincoveney**
Adds theming options for the `CollectionBrowser`'s `This collection is empty` message. There's two main changes:

_**Custom component**_

The `CollectionBrowser` now takes in a `EmptyContentComponent`:

```
type CollectionBrowserProps = {
    ...
	EmptyContentComponent?: (() => JSX.Element) | null;
	...
};
```

This will completely replace the `EmptyContent` message with the given component.

_**Theming**_
If the client still would like to use the same messages and icon, they're now supplied with:

```
collectionBrowser:
    ....
	emptyContent: {
      icon: {
        width: CSSProperties["width"];
        height: CSSProperties["width"];
      };
      title: {
        fontSize: CSSProperties["fontSize"];
      };
      subtitle: {
        fontSize: CSSProperties["fontSize"];
      };
    };
}
```

_**Open question:**_

- Should we add padding and margin? Right now they're not standard values so we can't just use a `gap` attribute to set the spacing.

### CODE REVIEW

File: enterprise/frontend/src/embedding-sdk/lib/theme/default-component-theme.ts [hunk 1]
```diff
@@ -38,6 +38,20 @@ export const DEFAULT_METABASE_COMPONENT_THEME: MetabaseComponentTheme = {
         hoverBackgroundColor: "var(--mb-color-brand)",
       },
     },
+    emptyContent: {
```
Comment (oisincoveney) [inline enterprise/frontend/src/embedding-sdk/lib/theme/default-component-theme.ts:41]: I think we should also add spacing here, but I don't know if we should basically add padding and margin options for all of these?

File: enterprise/frontend/src/embedding-sdk/lib/theme/default-component-theme.ts [hunk 2]
```diff
@@ -38,6 +38,20 @@ export const DEFAULT_METABASE_COMPONENT_THEME: MetabaseComponentTheme = {
...
+        height: "94",
+      },
+      title: {
+        fontSize: "1.5rem",
+        textColor: "var(--mb-color-text-dark)",
+      },
+      subtitle: {
+        fontSize: "1rem",
+        textColor: "var(--mb-color-text-medium)",
+      },
```
Comment (heypoom) [inline enterprise/frontend/src/embedding-sdk/lib/theme/default-component-theme.ts:53]: Curious why we have `textColor` here, as they can already override the `colors.text-dark` and `colors.text-medium` without changing this

File: enterprise/frontend/src/embedding-sdk/lib/theme/default-component-theme.ts (hunk 1 above)
Comment (heypoom) [reply enterprise/frontend/src/embedding-sdk/lib/theme/default-component-theme.ts:41]: I think we can keep it minimum until we have more customer needs or requests, that way we're not adding too many theme options.

File: enterprise/frontend/src/embedding-sdk/components/public/CollectionBrowser/CollectionBrowser.tsx [hunk 3]
```diff
@@ -37,6 +37,7 @@ type CollectionBrowserProps = {
   onClick?: (item: CollectionItem) => void;
   pageSize?: number;
   visibleEntityTypes?: UserFacingEntityName[];
+  EmptyContentComponent?: (() => JSX.Element) | null;
```
Comment (heypoom) [inline enterprise/frontend/src/embedding-sdk/components/public/CollectionBrowser/CollectionBrowser.tsx:40]: Wondering if `() => JSX.Element` should be `React.FC` instead

File: frontend/src/metabase/collections/components/CollectionContent/CollectionItemsTable.tsx [hunk 4]
```diff
@@ -157,6 +159,9 @@ export const CollectionItemsTable = ({
           !loading && !hasPinnedItems && unpinnedItems.length === 0;
 
         if (isEmpty && !loadingUnpinnedItems) {
+          if (isEmbeddingSdk && EmptyContentComponent) {
+            return <EmptyContentComponent />;
+          }
```
Comment (heypoom) [inline frontend/src/metabase/collections/components/CollectionContent/CollectionItemsTable.tsx:164]: Wondering if we can use `EmptyContentComponent` and not having to rely on a `isEmbeddingSdk`-specific check.

(we could even create a default EmptyContentComponent for the main Metabase app)

File: frontend/src/metabase/collections/components/CollectionEmptyState/CollectionEmptyState.tsx [hunk 5]
```diff
@@ -60,25 +59,32 @@ const DefaultCollectionEmptyState = ({
...
 };
 
 const CollectionEmptyIcon = (): JSX.Element => {
+  const theme = useMantineTheme();
   return (
-    <svg width="117" height="94" fill="none" xmlns="http://www.w3.org/2000/svg">
+    <svg
+      viewBox="0 0 117 94"
+      width={theme.other.collectionBrowser.emptyContent.icon.width}
+      height={theme.other.collectionBrowser.emptyContent.icon.height}
```
Comment (heypoom) [inline frontend/src/metabase/collections/components/CollectionEmptyState/CollectionEmptyState.tsx:84]: nit: we could destructure out `emptyContent` if you want to shorten this. same as the other 6 occurrences below

**Review by heypoom — APPROVED** [review]
Looks good!

File: enterprise/frontend/src/embedding-sdk/components/public/CollectionBrowser/CollectionBrowser.tsx (hunk 3 above)
Comment (oisincoveney) [reply enterprise/frontend/src/embedding-sdk/components/public/CollectionBrowser/CollectionBrowser.tsx:40]: That was the one! I couldn't think of the type so I went with this one. I'll change this now

File: frontend/src/metabase/collections/components/CollectionEmptyState/CollectionEmptyState.tsx (hunk 5 above)
Comment (oisincoveney) [reply frontend/src/metabase/collections/components/CollectionEmptyState/CollectionEmptyState.tsx:84]: I'm writing a quick utility to get theme keys so I can just move that to this PR

File: frontend/src/metabase/collections/components/CollectionContent/CollectionItemsTable.tsx (hunk 4 above)
Comment (heypoom) [reply frontend/src/metabase/collections/components/CollectionContent/CollectionItemsTable.tsx:164]: That sounds nice!

File: frontend/src/metabase/collections/components/CollectionEmptyState/CollectionEmptyState.tsx (hunk 5 above)
Comment (oisincoveney) [reply frontend/src/metabase/collections/components/CollectionEmptyState/CollectionEmptyState.tsx:84]: ok I'll put it in a different PR. typescript is being difficult

**Pull request merged**